### PR TITLE
Refine pre-series behavior and active-series linking

### PR DESCRIPTION
**Pull request by davidhouweling**
## Context
Streamer settings now flow correctly in production, and this PR focuses on pre-series correctness for both the individual viewer and stream overlays. The main goals were to improve initial series display values, carry forward NeatQueue start-time context, and ensure active series match-linking behaves correctly when the first eligible game appears.

## This PR

| Issue | Where the gap was | How this PR fixes it | Tests added/updated |
|---|---|---|---|
| Viewer series score should start at 0:0, not - | Pending active-series row used placeholder score values in the viewer render model | Pending active-series row now renders score as 0:0 with pre-series defaults | Updated assertions in `pages/src/components/individual-tracker/viewer/tests/viewer-render-model.test.ts` |
| Viewer missing-data formatting should be clean (- or N/A) | Series summary fallback formatting could show dense unknown templates | Added series-summary-specific fallback display as N/A for pre-series style summaries | Covered by updated viewer pending-row assertions and existing DO series tests |
| Pre-series start time should come from NeatQueue trigger | Started-at timestamp was not propagated through nudge/contract/view context | Added optional `startedAt` through nudge + view contracts, DO state/view context, mapper, and viewer model | Added startedAt assertion in `api/services/neatqueue/tests/neatqueue.test.ts` and active-series view assertion in `api/durable-objects/individual-tracker/tests/individual-tracker-do.test.ts` |
| Active series should include first eligible match without waiting for auto-detection threshold | Active-series grouping only surfaced when linked matches reached 2 | Added explicit active-series eligibility logic and changed active-series grouping threshold from 2+ to 1+ while preserving auto-detection behavior when no active series exists | Added regression test: `surfaces an active series group when only one active-series match is currently linked` in `api/durable-objects/individual-tracker/tests/individual-tracker-do.test.ts` |
| Stream overlay top team score should start at 0, not - | Overlay score display depended on raw upstream score strings | Added score normalization so invalid score values map to 0:0 in state render model | Added regression test in `pages/src/components/live-tracker/tests/state-render-model.test.ts` |
| Stream overlay Series score tab should read 0:0, not - | Same upstream score dependency as top score | Same normalization fix now feeds Series score tab values | Covered by the new state-render-model normalization test and existing overlay render tests |
| Information ticker should show pre-series info when series is pre-game | Pre-series mode used raw matches length, which could miss series pre-game state | Switched pre-series gating to `seriesData.matchIds` count and aligned panel/ticker conditions | Existing overlay suite remains passing (`pages/src/components/live-tracker/streamer-overlay/tests/streamer-overlay.test.tsx`) |

### Notes on active-series rule behavior
- Auto-detection behavior remains unchanged when there is no active series (existing grouping heuristics remain in place).
- When an active series exists, the “two consecutive custom games” requirement is intentionally bypassed so the first eligible match can be linked immediately.

### Validation
- Targeted suites passed:
  - `api/durable-objects/individual-tracker/tests/individual-tracker-do.test.ts`
  - `api/services/neatqueue/tests/neatqueue.test.ts`
  - `pages/src/components/individual-tracker/viewer/tests/viewer-render-model.test.ts`
  - `pages/src/components/live-tracker/tests/state-render-model.test.ts`
  - `pages/src/components/live-tracker/streamer-overlay/tests/streamer-overlay.test.tsx`
- `npm run typecheck` passed.
- Full `npm run done` reached green format/typecheck/lint and then hit the existing `test:changed` script edge case (`vitest related` with no related files), while explicit targeted suites above passed.

## Subsequent Work
- Add a focused overlay regression test that explicitly validates pre-series ticker behavior when `seriesData.matchIds` is empty while raw matches may still be present.

### CODE REVIEW

File: api/durable-objects/individual-tracker/individual-tracker-do.ts [hunk 1]
```diff
@@ -321,6 +329,27 @@ function getSeriesSummariesForView(
...
+  const expectedTeamCounts = getExpectedSeriesTeamCounts(activeSeries.teams);
+  if (expectedTeamCounts == null) {
+    if (summary.isMatchmaking) {
+      return false;
+    }
+
+    return true;
+  }
+
+  return hasExpectedSeriesTeamCounts(summary, expectedTeamCounts);
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply api/durable-objects/individual-tracker/individual-tracker-do.ts:350]: Fixed in 2c08c99c. Matchmaking matches are now always ineligible for active-series linking, so they correctly end the active series even when team counts match.

File: pages/src/components/live-tracker/streamer-overlay/streamer-overlay.tsx [hunk 2]
```diff
@@ -76,6 +76,8 @@ function NeatQueueStreamerOverlay({
   settingsUi,
 }: NeatQueueStreamerOverlayProps): React.ReactElement {
   const SharedStreamerOverlay = useMemo(() => createStreamerOverlaySection(), []);
+  const seriesMatchCount = neatQueueState.seriesData?.matchIds.length ?? neatQueueState.matches.length;
+  const isPreSeriesMode = seriesMatchCount === 0;
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply pages/src/components/live-tracker/streamer-overlay/streamer-overlay.tsx:80]: Fixed in 2c08c99c. The live tracker now passes seriesData match count to the shared overlay, keeping pre-series mode and ticker synchronization consistent.

File: api/durable-objects/individual-tracker/individual-tracker-do.ts [hunk 3]
```diff
@@ -2368,7 +2400,7 @@ export class IndividualTrackerDO implements DurableObject, Rpc.DurableObjectBran
     const activeSeriesMatchIds = state.activeSeries?.matchIds ?? [];
     const activeSeriesMatchIdSet = new Set(activeSeriesMatchIds);
     const groupings =
-      activeSeriesMatchIds.length >= 2
+      activeSeriesMatchIds.length > 0
         ? [activeSeriesMatchIds, ...autoGroupings.filter((g) => !g.some((id) => activeSeriesMatchIdSet.has(id)))]
         : autoGroupings;
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply api/durable-objects/individual-tracker/individual-tracker-do.ts:2405]: Fixed in 724f2697. Active series with one known linked match now renders as the active series row in the viewer instead of showing a pending pre-series row plus standalone match.

File: api/services/neatqueue/neatqueue.ts [hunk 4]
```diff
@@ -701,6 +702,20 @@ export class NeatQueueService {
...
+      .find((value) => {
+        return value.trim() !== "";
+      });
+
+    if (candidateTimestamp != null && !Number.isNaN(Date.parse(candidateTimestamp))) {
+      return new Date(candidateTimestamp).toISOString();
+    }
+
+    return new Date().toISOString();
+  }
```
[the remark this answers is not in the document]
Comment (davidhouweling) [reply api/services/neatqueue/neatqueue.ts:718]: Fixed in 724f2697. Series startedAt now uses the earliest valid player timestamp in the payload and ignores invalid/empty timestamps instead of falling back to current time when valid values exist.
